cuda_core/examples/simple_multi_gpu_example.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+from cuda.core.experimental import Device
+from cuda import cuda, cudart, nvrtc
+from cuda.core.experimental import LaunchConfig, launch
+from cuda.core.experimental import Program
+from cuda.core.experimental._memory import Buffer
+
+import cupy as cp
+
+dtype = cp.float32
+size = 50000
+
+# Set GPU0
+dev0 = Device(0)
+dev0.set_current()
+stream0 = dev0.create_stream()
+
+# allocate memory to GPU0
+a = cp.random.random(size, dtype=dtype)
+b = cp.random.random(size, dtype=dtype)
+c = cp.empty_like(a)
+
+# Set GPU1
+dev1 = Device(1)
+dev1.set_current()
+stream1 = dev1.create_stream()
+
+# allocate memory to GPU1
+x = cp.random.random(size, dtype=dtype)
+y = cp.random.random(size, dtype=dtype)
+z = cp.empty_like(a)
+
+# compute c = a + b
+code_add = """
+template<typename T>
+__global__ void vector_add(const T* A,
+                           const T* B,
+                           T* C,
+                           size_t N) {
+    const unsigned int tid = threadIdx.x + blockIdx.x * blockDim.x;
+    for (size_t i=tid; i<N; i+=gridDim.x*blockDim.x) {
+        C[tid] = A[tid] + B[tid];
+    }
+}
+"""
+
+# compute c = a - b
+code_sub = """
+template<typename T>
+__global__ void vector_sub(const T* A,
+                           const T* B,
+                           T* C,
+                           size_t N) {
+    const unsigned int tid = threadIdx.x + blockIdx.x * blockDim.x;
+    for (size_t i=tid; i<N; i+=gridDim.x*blockDim.x) {
+        C[tid] = A[tid] - B[tid];
+    }
+}
+"""
+
+prog_add = Program(code_add, code_type="c++")
+mod_add = prog_add.compile(
+    "cubin",
+    options=("-std=c++17", "-arch=sm_" + "".join(f"{i}" for i in dev0.compute_capability),),
+    name_expressions=("vector_add<float>",))
```

Comment:
nits:
1. construct the arch string explicitly
2. remove `name_expressions`
```suggestion
arch0 = "".join(f"{i}" for i in dev0.compute_capability)
prog_add = Program(code_add, code_type="c++")
mod_add = prog_add.compile(
    "cubin",
    options=("-std=c++17", "-arch=sm_" + arch0,))
```